ZeCamp/ZeCamp/Sources/Schedule/ScheduleScreen.swift

```diff
@@ -56,7 +58,7 @@ struct ScheduleScreen {
         
         
         
-        let delegate = ScheduleDelegate()
+        let delegate = ScheduleDelegate(timeSlots)
```

Comment:
The `dataSource` and the `delegate` can be the same class. In this way you don't need to keep an instance of the timeslots and pass it into the two places separately. This will make it easier to filter the timeslots by day of week. To do this you need to implement `UITableViewDelegate` in `ScheduleDataSource`.